### PR TITLE
chore: set up TypeScript auto-formatting

### PR DESCRIPTION
**Pull request by Steborino**
## tl;dr:

This sets up auto-formatting for TypeScript. If using Visual Studio Code, users will be prompted to install a plugin which will automatically format the code based on the configuration set in `.prettierrc.js`. I based the initial config off of the one from [`@google/gts`](https://github.com/google/gts)—I may be biased, but Google generally has good taste.

You can feel free to modify this config as you see fit; no matter what styles you choose, the codebase will remain consistent with your choices.

Formatting is also enforced at the linter level. If users do not install the VSCode plugin, they can run the `fix` NPM script to auto-format the codebase.

## Details

- Sets up [Prettier](https://github.com/prettier/prettier).
  - Add `.prettierrc.js` to configure initial code style preferences (can be changed whenever).
  - Add plugin to ESLint.
  - Set up VSCode workspace extension recommendations so users are prompted to install Prettier extension.
  - Set up VSCode workspace settings so Prettier is run whenever files are being changed.
- Adds new npm script (e.g. in case you didn't feel like installing the Prettier extension):

    ```
    npm run fix
    ```
- Marks a bunch of new lint errors as warnings, for now, with TODOs. Lint errors are pre-existing, but may have been surfaced with installation of Prettier ESLint plugin.
- In final commit (`154582b`), runs new `fix` script to auto-format the entire codebase.

### CODE REVIEW

**Comment by Steborino** [comment]
Merged things around so there ain't no merge conflicts now. 😎
